### PR TITLE
update google GTest to 1.10.0

### PR DESCRIPTION
**Pull request by TorreZuk**
* match rocBLAS and fixes deprecated use warnings

### CODE REVIEW

**Review by daineAMD — APPROVED** [review]
Looks like an unrelated static-libs CI error? Looks fine otherwise!

**Comment by TorreZuk** [comment]
Thanks,  I'll wait until after you merge master to develop PR, this can go in next week.  Couple other CI issues as you mention are unrelated I think

**Pull request merged**
